portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java

```diff
@@ -62,6 +90,18 @@ public void doGet(HttpServletRequest request,
     response.getWriter().println(convertToJson(messages));
   }
 
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameterWithDefault(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
IIRC, I think you have added this method elsewhere?
Please consider create a helper class that contains these reusable methods rather than duplicate code.